### ISSUE TITLE
Add "clear filters" button to map and change "search this area" to "search here"

### ISSUE THREAD

**Issue by jesseallhands** [Couchers-org/couchers#5275]
### Problem
Right now it is very difficult for users to "reset" all search filters so that all pins are shown on the map again after they have performed a search with any filter. A common problematic situation is this:

1. User performs search on map
2. Limited number of users are shown on the screen due to the search
3. User wants to expand their search beyond the limits of their initial search (eg. they want to see everyone in the area now).
4. The user zooms out and very few pins are shown, leading them to think there are very few users on Couchers

### Solution
The new buttons should look like this on mobile (update- change "show all" to "clear filters"):
![image](https://github.com/user-attachments/assets/c34c3bd2-a899-40cf-94f2-23c15b7195e5)
- [x] If the user has applied a filter, display a "clear filters" button which reverts all search parameters so that all pins are shown on the map. If no filters are applied, show disabled button. 
- [x] To ensure there is enough space on mobile, the button "Search in this area" should be changed to "Search here" 
- [x] Remove the spinner to give more room. The spinner is being added as an overlay to the map in #5026 
- [x] Update border-radius and margin of buttons (so there is enough space in between the buttons)
- [x] Test

**Comment by bakeiro** [comment]
does "show all" means "clear filters"?

**Comment by jesseallhands** [comment]
> does "show all" means "clear filters"?

Yes, "show all" is more intuitive. Some users might not consider entering a location as "a filter" (filter is kind of a technical term for basic users) so it's better to use "show all" since that's what the button does - shows all pins/results.

**Comment by graceegal** [comment]
@bakeiro are you tackling this issue? I see it's unassigned, so if not I can pick it up :)

**Comment by jesseallhands** [comment]
@graceegal I think David is not doing PRs right now since he got a new job so you can def grab this one!

**Comment by graceegal** [comment]
> @graceegal I think David is not doing PRs right now since he got a new job so you can def grab this one!

Okay great! Thank you!

**Comment by bakeiro** [comment]
1. what when we implement pagination and user it's for example in page 12, and then clicking by mistake 'show all' loses the page and all the filters?
2. I think that "The user zooms out and very few pins are shown, leading them to think there are very few users on Couchers" is not true, since if we show the total number of users [https://github.com/Couchers-org/couchers/issues/4991] plus the number of pages that shouldn't happen...

maybe this deserves more UX research? plus I feel that returning only a subset after showing all the results first is confusing for the users...

**Comment by jesseallhands** [comment]
@bakeiro A button to reset filters is common practice. This is an urgent issue because the current situation makes the map search on mobile nearly unusable after the first search unless the user force refreshes the screen. I encourage you to try it out yourself on mobile under several use-cases. We can gather user feedback after the addition of the button and iterate if necessary.

**Comment by bakeiro** [comment]
> @bakeiro A button to reset filters is common practice. This is an urgent issue because the current situation makes the map search on mobile nearly unusable after the first search unless the user force refreshes the screen. I encourage you to try it out yourself on mobile under several use-cases. We can gather user feedback after the addition of the button and iterate if necessary.

I agree on clearing filters, that's totally fine, the thing is

if the user clicks by mistake, since both buttons are next to each other, he would lose the page info plus the filters, and if for example the user is on the page 10, in mobile you have to move between pages 1 by 1, so to go back to the page where the user was I think it would be very uncomfortable...

if we implement pagination, users no longer need to reload the page?, since they can see all the results if they move between pages.

if the user thinks that there are too few so it can be indicated in the ticket that I mentioned previously.

About the map being unusable (which I agree) in my experience it's because the map is so slow, the main reason is showing all the users on the map, I have an old phone, I can ensure you that makes the page totally unusable, I'm wondering whether this button makes the page usable for old phones as well...

I had other ideas but requires more re-search and work...


edit:
> We can gather user feedback after the addition of the button and iterate if necessary

totally agree on this

>the current situation makes the map search on mobile nearly unusable after the first search unless the user force refreshes the screen

totally agree, that's why I said returning a subset after the first search is confusing

**Comment by bakeiro** [comment]
I also though:
- Shouldn't we only show this button *only* after the user applies a filter/search in an area? otherwise user will see a button, that clicking on it when we already display all the users don't do nothing [which will be confusing...] maybe showing after searching makes more sense?

btw, I don't want to be the blocker of this ticket [I am being to picky? xd] but I feel this needs more research, maybe other people have other opinion here?

@nabramow @aapeliv

**Comment by jesseallhands** [comment]
>Shouldn't we only show this button only after the user applies a filter/search in an area?

Yes! Good point. I updated the ticket.

**Comment by nabramow** [comment]
@jesseallhands @bakeiro TBH I think it makes more sense to me to have the existing filters button be a highlighted color when there are filters active and then within that dialog have a button "Clear all filters" and that will also change the color back to not highlighted. Having an additional button seems eh.

TBH long-term seems better to not have filters in a pop up anyway, but rather together with the search field. But that's a bigger ticket.

**Comment by jesseallhands** [comment]
>TBH long-term seems better to not have filters in a pop up anyway, but rather together with the search field. But that's a bigger ticket.

I totally agree with this. This ticket is just a temporary fix.

**Comment by nabramow** [comment]
If we're going to add an additional button as a stop gap though, what if we add one with text "Clear filters" rather than show all.

That seems more common practice?

@jesseallhands

**Comment by bakeiro** [comment]
> [@jesseallhands](https://github.com/jesseallhands) [@bakeiro](https://github.com/bakeiro) TBH I think it makes more sense to me to have the existing filters button be a highlighted color when there are filters active and then within that dialog have a button "Clear all filters" and that will also change the color back to not highlighted. Having an additional button seems eh.
> 
> TBH long-term seems better to not have filters in a pop up anyway, but rather together with the search field. But that's a bigger ticket.

that's what I found in the Dom's mockups... the point is that here this button deletes the location as well, not only the filters... [plus pagination in the future if I'm not wrong]

**Comment by nabramow** [comment]
@bakeiro @jesseallhands I'm adding this to my agenda for Tuesday to make a decision there so poor Gracee doesn't have to read anymore of our comments, hah.
